lib/matplotlib/axes/_base.py

```diff
@@ -3924,34 +3924,36 @@ def twinx(self):
         ax2.yaxis.tick_right()
         ax2.yaxis.set_label_position('right')
         ax2.yaxis.set_offset_position('right')
+        ax2.set_autoscalex_on(self.get_autoscalex_on())
         self.yaxis.tick_left()
         ax2.xaxis.set_visible(False)
         ax2.patch.set_visible(False)
         return ax2
 
     def twiny(self):
         """
-        Create a twin Axes sharing the yaxis
+        Create a twin Axes instance sharing the y-axis with self
 
-        Create a twin of Axes for generating a plot with a shared
-        y-axis but independent x-axis.  The x-axis of self will have
-        ticks on bottom and the returned axes will have ticks on the
-        top.
+        Create a twin Axes instance for generating a plot with a shared
+        y-axis but independent x-axis.  The x-ticks on the returned Axes will
+        be on the top, opposite those of the original Axes. The y-axis
+        autoscale setting will be inherited from the parent Axes. To ensure
+        tick marks of both axis align, see `~matplotlib.ticker.LinearLocator`
 
         Returns
         -------
-        Axis
-            The newly created axis
+        Axes
+            The newly created Axes instance
 
         Notes
-        ------
+        -----
         For those who are 'picking' artists while using twiny, pick
         events are only called for the artists in the top-most axes.
```

Comment:
The `Axes` are rendered in some order (determined by the z-order and sorted in the `Figure` draw method).  Which ever one draws last (has highest z-order) is the one that gets the pick event.